libs/backend/feature/table/src/lib/round/winner-determiner/util/rank.ts

```diff
@@ -151,21 +151,40 @@ export function rankHand(player: PlayerWithHand): RankHandReponse {
             ) * 2;
     }
 
-    // Add the rank of the highest card to the score, which acts as the kicker to break ties
+    // Add the rank of the highest card(s) to the score, which act as the kicker(s) to break ties.
+    // Every hand must form 5 cards, so count the number of cards in the main hand and the excess
+    // become the kickers we have to consider.
     if (
         category === 'four of a kind' ||
         category === 'three of a kind' ||
         category === 'two pairs' ||
         category === 'pair' ||
         category === 'high card'
     ) {
-        score += Math.max(
-            ...Object.keys(ranks)
-                .filter((key) => ranks[key as Rank] === 1)
-                .map(Number),
-        );
+        // the number of kickers to consider for each category
+        const kickersCountMap: Record<string, number> = {
+            'four of a kind': 1,
+            'three of a kind': 2,
+            'two pairs': 1,
+            'pair': 3,
+            'high card': 5
+        };
+
+        let kickersCount = kickersCountMap[category];
```

Comment:
```suggestion
        const kickersCount = kickersCountMap[category];
```

Reply:
Good catch, fixed for this and https://github.com/broology/poker-moons/pull/308/files#r1220488535